remove dev_settings from source control and add to .gitignore

stuartjennings-up opened the pull request.

stuartjennings-up committed:
remove dev_settings from source control and add to .gitignore

craiga reviewed (commented):
Why are we removing `dev_settings.py`?

At some point, we'll review why we need all of these different settings files and perhaps move to a system like http://stackoverflow.com/a/15325966/1852024.

stuartjennings-up committed:
add dev_settings_default without orcid keys to be used as template for new devs

craiga commented:
@stuartjennings-up Where are we at with this PR?

stuartjennings-up commented:
@craiga for some reason this was included in the other PR that you just approved so I think I can just remove this one. I've added dev_settings.py back in but added it to .gitignore and replaced it with dev_settings_default as you suggested
